components/tcf/secondLayer/src/index.js

```diff
@@ -133,26 +196,46 @@ export default function TcfSecondLayer({
           onAcceptAll={handleAcceptAll}
           onRejectAll={handleRejectAll}
         />
-        {state && state.purposes && (
-          <PurposeGroup
-            name="Purposes"
-            descriptionField="description"
-            descriptions={purposesVLState}
-            state={state.purposes}
-            onConsentsChange={handlePurposesConsentsChange}
-            onLegitimateInterestsChange={handlePurposesLIChange}
-          />
-        )}
-        {state && state.vendors && (
-          <PurposeGroup
-            name="Vendors"
-            descriptionField="name"
-            descriptions={vendorsVLState}
-            state={state.vendors}
-            onConsentsChange={handleVendorsConsentsChange}
-            onLegitimateInterestsChange={handleVendorsLIChange}
-          />
-        )}
+        {state &&
+          state.purposes &&
+          vendorListState &&
+          vendorListState.purposes && (
+            <PurposeGroup
+              name="Purposes"
+              descriptionField="description"
+              descriptions={vendorListState.purposes}
+              state={state.purposes}
+              onConsentsChange={handlePurposesConsentsChange}
+              onLegitimateInterestsChange={handlePurposesLIChange}
+              hasLegitimateInterest
+            />
+          )}
+        {state &&
+          state.vendors &&
+          vendorListState &&
+          vendorListState.vendors && (
```

Comment:
NBC. You can use if you want
state?.vendors && vendorListState?.vendors && ...

Reply:
done, thanks